app/views/courses/revisions.json.jbuilder

```diff
@@ -1,11 +1,15 @@
 # frozen_string_literal: true
 
 json.course do
-  json.course_specific_revision_ids @course.tracked_revisions
-                                           .order(date: :desc).limit(@limit).pluck(:id)
-  json.revisions(@course.recent_revisions
-                        .eager_load(:user, :wiki).includes(article: :wiki)
-                        .order(date: :desc).limit(@limit)) do |rev|
+  revisions = if @course_scoped
+                @course.tracked_revisions.order(date: :desc).limit(@limit)
```

Comment:
I think you want to use the same `eager_load` in the query here; I believe that prevents N+1 queries when rendering the json for each individual revision.

Reply:
Done!